src/blocks/scratch3_control.js

```diff
@@ -7,6 +7,12 @@ class Scratch3ControlBlocks {
          * @type {Runtime}
          */
         this.runtime = runtime;
+
+        /**
+         * The "counter" block value. For compatibility with 2.0.
+         * @type {number}
+         */
+        this._counter = 0;
```

Comment:
@cwillisf Does adding this here complicate extension-ifying control blocks? Having this stored here all the time seems weird to me given that these are not official blocks and [are not used very often](https://github.com/LLK/scratch-vm/issues/355#issuecomment-379418752), but I'm not sure what's a better way to keep track of this value. Having it be a private variable is definitely a good step towards eliminating confusion, but is there a better place to store this info? Maybe in the runtime?

Reply:
I wonder if it would make sense to define these blocks in a separate package, like "Legacy Control" or something. Then this could be a private variable in that blocks package and, in the future, that extension.
The good news is that all of this is fairly self contained, in that `_counter` is only manipulated by these new block implementation functions, so we can move it fairly easily later. My inclination would be to go with this as-is but talk about our options SOON (next product meeting?) and, if we decide to move these opcodes, try to get that done before we have people saving projects with these blocks :)
A third option is to add support for extensions which add blocks to existing categories. I don't think that would be difficult but I'm not sure it's a good idea.

Reply:
@cwillisf Thanks for the response! I think your first inclination of merge this as is but discuss the possibility of moving these SOON sounds like the best option (especially the bit about saving projects).

Reply:
For reference, I noticed that the "sensing" blocks also store data like this. In particular, the "answer" block's output is [stored](https://github.com/LLK/scratch-vm/blob/d177f4cd0c316653b83c7547c001566523d7f374/src/blocks/scratch3_sensing.js#L12-L16) (and set by "ask"), which I figured is pretty much the same thing as storing the counter value.